File: kinotic-js/workspace/packages/os-api/src/api/model/grind/StepCompletion.ts
import { StoreType } from '@/api/model/grind/StoreType'

/**
 * The value of a STEP_COMPLETED Result, describing what the completed step stored in the
 * job scope when it finished.
 */
export class StepCompletion {

    /**
     * How the step stored its result, NONE if it stored nothing.
     */
    public storeType: StoreType = StoreType.NONE

    /**
     * The name the step's result was stored under, or null if the step stored nothing
     * or stored under generated names.
     */
    public storedName: string | null = null

    /**
     * The stored value. Always null over the monitoring stream - the produced object stays
     * in the executing process.
     */
    public storedValue: any = null

}

File: kinotic-grind/src/main/java/org/kinotic/grind/api/model/StepCompletion.java
package org.kinotic.grind.api.model;

/**
 * The value of a {@link ResultType#STEP_COMPLETED} result, describing what the completed step
 * stored in the {@code JobContext} when it finished.
 *
 * @param storeType   How the step stored its result, {@link StoreType#NONE} if it stored nothing.
 * @param storedName  The name the step's result was stored under, or null if the step stored nothing
 *                    or stored under generated names.
 * @param storedValue The value the step stored in the {@code JobContext}, or null if the step stored nothing.
 */
public record StepCompletion(StoreType storeType, String storedName, Object storedValue) {

}
